models/vrp.rb

```diff
@@ -174,17 +177,22 @@ def self.check_consistency(hash)
         raise OptimizerWrapper::DiscordantProblemError, 'All matrices should have at least maximum(point[:matrix_index]) number of rows and columns' if matrix_not_big_enough
       end
 
+      # services consistency
+      if (hash[:services] + hash[:shipments]).any?{ |s| (s[:minimum_lapse] || 0) > (s[:maximum_lapse] || 2**32) }
+        raise OptimizerWrapper::DiscordantProblemError.new('Minimum lapse can not be bigger than maximum lapse')
+      end
+
       # shipment position consistency
       forbidden_position_pairs = [[:always_middle, :always_first], [:always_last, :always_middle], [:always_last, :always_first]]
-      hash[:shipments]&.each{ |shipment|
+      hash[:shipments].each{ |shipment|
```

Comment:
what about the shipments that are given with relations? hmm this "duality" of  model:shipment vs relation:shipment is more problematic than we realised I think.